src/compute-engine/global-types.ts

```diff
@@ -1201,18 +1226,25 @@ export interface BoxedExpression {
    *
    * Equivalent to `expr.N().valueOf()`.
    *
+   * For functions, will only return non-undefined (i.e., compute the value) if the function is pure.
```

Comment:
I commented above about this, but I'm not sure why that would be the case. What would be more useful?

Reply:
I'm also wondering if there isn't some confusion between a pure function and a pure function expression... That is `Sin` is a pure function, `["Sin", "Pi"]` is a pure function expression, `Random` is an impure function, `["Sin", ["Random"]]` is an impure function expression, `["Sin", "x"]` is a pure expression iff `"x"` is a constant.